pkg/services/validator/oci.go

```diff
@@ -46,6 +51,50 @@ func readOciPluginRules(vc *components.ValidatorConfig, _ *cfg.TaskConfig, kClie
 	return nil
 }
 
+// configureAuth prompts the user to configure their OCI registry authentication details.
+func configureAuth(r *plug.OciRegistryRule) error {
+	r.Auth.SecretName = nil
+
+	authType, err := prompts.Select("Authentication type", []string{basicAuth, ecrAuth})
+	if err != nil {
+		return err
+	}
+
+	if authType == basicAuth {
+		if r.Auth.Basic == nil {
```

Comment:
Should we explicitly set `r.Auth.ECR = nil` here?

Reply:
we dont _need_ to because of the way the creds are read. they prioritize the secret, then basic, then ecr. But enforcing it to be nil here wouldn't hurt in case the order of priority changes at some point